pkg/test/executor/test_runner.go

```diff
@@ -38,69 +38,57 @@ func RunTestCase(
 
 	fsr, c := setup.SetupBacalhauRepoForTesting(t)
 	stack := testutils.Setup(ctx, t, fsr, c, devstackOptions.Options()...)
-	executor, err := stack.Nodes[0].ComputeNode.Executors.Get(ctx, spec.EngineSpec.Engine().String())
+	executor, err := stack.Nodes[0].ComputeNode.Executors.Get(ctx, job.Task().Engine.Type)
 	require.NoError(t, err)
 
 	isInstalled, err := executor.IsInstalled(ctx)
 	require.NoError(t, err)
 	require.True(t, isInstalled)
 
-	prepareStorage := func(getStorage scenario.SetupStorage) []model.StorageSpec {
+	prepareStorage := func(getStorage scenario.SetupStorage) []*models.InputSource {
 		if getStorage == nil {
-			return []model.StorageSpec{}
+			return []*models.InputSource{}
 		}
 
 		storageList, stErr := getStorage(ctx)
 		require.NoError(t, stErr)
 
-		for _, storageSpec := range storageList {
-			inputSource, err := legacy.FromLegacyStorageSpecToInputSource(storageSpec)
+		for _, input := range storageList {
+			strger, err := stack.Nodes[0].ComputeNode.Storages.Get(ctx, input.Source.Type)
 			require.NoError(t, err)
 
-			strger, err := stack.Nodes[0].ComputeNode.Storages.Get(ctx, storageSpec.StorageSource.String())
-			require.NoError(t, err)
-			hasStorage, stErr := strger.HasStorageLocally(ctx, *inputSource)
+			hasStorage, stErr := strger.HasStorageLocally(ctx, *input)
 			require.NoError(t, stErr)
+
 			require.True(t, hasStorage)
 		}
 
 		return storageList
 	}
 
-	spec.Inputs = prepareStorage(testCase.Inputs)
-	spec.Outputs = testCase.Outputs
-	spec.PublisherSpec = model.PublisherSpec{
-		Type: model.PublisherLocal,
-	}
-	spec.Deal = model.Deal{Concurrency: testNodeCount}
-
-	job := model.Job{
-		Metadata: model.Metadata{
-			ID:        "test-job",
-			ClientID:  "test-client",
-			CreatedAt: time.Now(),
-			Requester: model.JobRequester{
-				RequesterNodeID: "test-owner",
-			},
-		},
-		Spec: spec,
-	}
-
-	newJob, err := legacy.FromLegacyJob(&job)
-	require.NoError(t, err)
-	execution := mock.ExecutionForJob(newJob)
-	execution.AllocateResources(newJob.Task().Name, models.Resources{})
-
-	// Since we are submitting jobs directly to the executor, we need to
-	// transform wasm storage specs, which is currently handled by the
-	// requester's endpoint
-	jobTransformers := []jobtransform.PostTransformer{
-		jobtransform.NewWasmStorageSpecConverter(),
-	}
-	for _, transformer := range jobTransformers {
-		_, err = transformer(ctx, newJob)
-		require.NoError(t, err)
-	}
```

Comment:
this is no longer needed?

Reply:
NewWasmStorageSpecConverter converts the entry and import modules of wasm engine spec from the legacy storage spec to input source definitions. Since we aren't using the legacy spec anymore this shouldn't be required.